invenio_oauthclient/handlers.py

```diff
@@ -350,6 +346,14 @@ def signup_handler(remote, *args, **kwargs):
                 account_setup=account_setup
             )
 
+        # Authenticate the user
+        if not oauth_authenticate(remote.consumer_key, user,
+                                  require_existing_link=False,
+                                  remember=current_app.config[
+                                      'OAUTHCLIENT_REMOTE_APPS']
+                                  [remote.name].get('remember', False)):
+            return redirect(url_for("security.login"))
```

Comment:
This is the use case:
 - user try to login with orcid
 - orcid successfully return orcid id (buy without email)
 - the app now can register the user (but not activate him), that's why `oauth_authenticate` return `False`.
 - In that case, the app redirect the user somewhere, waiting that the user confirms his email (I though that return `not authorized` error is not really a good message for the user because actually he didn't do anything wrong)

Any improvement will be welcome :)